best-practices/sdk-principles/kotlin.md

```diff
@@ -0,0 +1,73 @@
+# Client software (SDK) principles and best practice for Kotlin (and any other JVM language)
+
+Background and product stratagy context can be found in [SDK principles for JS](js.md)
+
+Kotlin is the primary development language for any JVM-based SDKs (Java, Scala, Groovy, Clojure, etc). Our main focus is to provide an idiomatic Kotlin API for our customers. However, upon request, we will also offer a Java package that does not use coroutines or extension functions, making it easier to consume from Java (similar to [apollo java support](https://github.com/apollographql/apollo-kotlin-java-support)).
```

Comment:
Does this mean we'll already have a barebones Java package at the ready, or this is something we'd work on if we were asked for it?

Reply:
I believe we shouldn't provide barebones Java package by default. For our core SDK - Ably Java we of course will provide it, but for new products - I don't think so. Most of our JVM SDK users are from Android, and Kotlin is main language for Android. And also it's relatively easy to create wrappers for coroutines in separate Kotlin module. You can even use Kotlin files inside your Java project (it adds some compilation time penalties, but it's very easy to do and support)